Makefile

```diff
@@ -69,7 +69,7 @@ lint-go:
 lint-git: fetch-git
 	@COMMIT_SHA=`git rev-parse $(OASIS_CORE_GIT_ORIGIN_REMOTE)/$(RELEASE_BRANCH)` && \
 	echo "Running gitlint for commits from $(OASIS_CORE_GIT_ORIGIN_REMOTE)/$(RELEASE_BRANCH) ($${COMMIT_SHA:0:7})..."; \
-	gitlint --commits $(OASIS_CORE_GIT_ORIGIN_REMOTE)/$(RELEASE_BRANCH)...HEAD
+	gitlint --commits $(OASIS_CORE_GIT_ORIGIN_REMOTE)/$(RELEASE_BRANCH)..HEAD
```

Comment:
From `git-diff` manual:
```
git diff [<options>] <commit> <commit> [--] [<path>...]
   This is to view the changes between two arbitrary <commit>.

git diff [<options>] <commit>..<commit> [--] [<path>...]
   This is synonymous to the previous form. If <commit> on one side is
   omitted, it will have the same effect as using HEAD instead.

git diff [<options>] <commit>...<commit> [--] [<path>...]
   This form is to view the changes on the branch containing and up to
   the second <commit>, starting at a common ancestor of both
   <commit>. "git diff A...B" is equivalent to "git diff $(git
   merge-base A B) B". You can omit any one of <commit>, which has the
   same effect as using HEAD instead.
```

In case a branch is rebased on top of the current `${RELEASE_BRANCH}` (e.g. `master` branch), using `<commit>..<commit` or `<commit>...<commit>` doesn't make a difference.

However, in case a branch is not up-to-date, then using the `<commit>...<commit>` form will compare the current branch's `HEAD` with the common ancestor of the `${RELEASE_BRANCH}` and the current branch, i.e. the commit from which the current branch was branched from.
In my opinion, that makes more sense since one wants to see `gitlint`'s output for the commits he made in his current branch.

Reply:
Thanks for looking into this. `gitlint` internally uses `git rev-list` rather than `git diff`, with `git rev-list`'s documentation stating:

```
       Thus, the following command:

                   $ git rev-list foo bar ^baz

       means "list all the commits which are reachable from foo or bar, but
       not from baz".

       A special notation "<commit1>..<commit2>" can be used as a short-hand
       for "^'<commit1>' <commit2>". For example, either of the following may
       be used interchangeably:

                   $ git rev-list origin..HEAD
                   $ git rev-list HEAD ^origin

       Another special notation is "<commit1>...<commit2>" which is useful
       for merges. The resulting set of commits is the symmetric difference
       between the two operands. The following two commands are equivalent:

                   $ git rev-list A B --not $(git merge-base --all A B)
                   $ git rev-list A...B
```

Reply:
Oh, I see, thanks for looking in how gitlint implements this.

In this case, the correct way is to use `<commit>..<commit>`.

Can you add a comment above the `lint-git` target noting that gitlint uses `git rev-list` internally?

Reply:
yeah